libs/cache/CMakeLists.txt

```diff
@@ -5,150 +5,64 @@
 
 cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
 
-project(HPX.cache CXX)
+set(module_name cache)
+string(TOUPPER ${module_name} upper_module_name)
 
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-
-include(HPX_AddDefinitions)
-include(HPX_Option)
-
-hpx_option(HPX_CACHE_WITH_TESTS
-  BOOL
-  "Build HPX cache module tests. (default: ON)"
-  ON ADVANCED
-  CATEGORY "Modules")
-
-hpx_option(HPX_CACHE_WITH_DEPRECATION_WARNINGS
-  BOOL
-  "Enable warnings for deprecated facilities. (default: ${HPX_WITH_DEPRECATION_WARNINGS})"
-  ${HPX_WITH_DEPRECATION_WARNINGS} ADVANCED
-  CATEGORY "Modules")
-
-if(HPX_CACHE_WITH_DEPRECATION_WARNINGS)
-  hpx_add_config_define_namespace(
-      DEFINE HPX_CACHE_HAVE_DEPRECATION_WARNINGS
-      NAMESPACE CACHE)
-endif()
-
-# Added in 1.4.0
-hpx_option(HPX_CACHE_WITH_COMPATIBILITY_HEADERS
-  BOOL
-  "Enable compatibility headers for old headers"
-  ON ADVANCED
-  CATEGORY "Modules")
+project(HPX.${module_name} CXX)
 
-if(HPX_CACHE_WITH_COMPATIBILITY_HEADERS)
-  hpx_add_config_define_namespace(
-      DEFINE HPX_CACHE_HAVE_COMPATIBILITY_HEADERS
-      NAMESPACE CACHE)
-endif()
-
-message(STATUS "cache: Configuring")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-set(headers
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/local_cache.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/lru_cache.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/entries/entry.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/entries/fifo_entry.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/entries/lfu_entry.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/entries/lru_entry.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/entries/size_entry.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/policies/always.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/statistics/local_full_statistics.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/statistics/local_statistics.hpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/include/hpx/cache/statistics/no_statistics.hpp
+set(source_root ${CMAKE_CURRENT_SOURCE_DIR}/src)
+set(header_root ${CMAKE_CURRENT_SOURCE_DIR}/include)
+set(compat_header_root ${CMAKE_CURRENT_SOURCE_DIR}/include_compatibility)
```

Comment:
These three variables are always defined the same (for all modules). Should we use those as defaults inside add_hpx_module if they are not specified explicitly?